contracts/BootstrapRewards.sol

```diff
@@ -65,35 +73,34 @@ contract BootstrapRewards is IBootstrapRewards, ContractRegistryAccessor {
         return lastPayedAt;
     }
 
-    function assignRewards() external {
-        _assignRewards();
+    function assignRewards(address[] calldata generalCommittee, bool[] calldata compliance) external onlyElectionsContract {
+        _assignRewards(generalCommittee, compliance);
     }
 
-    function _assignRewards() private {
-        (address[] memory generalCommittee,) = getGeneralCommitteeContract().getCommittee();
-        _assignRewardsToCommittee(generalCommittee, generalCommitteeAnnualBootstrap);
-
-        (address[] memory complianceCommittee,) = getComplianceCommitteeContract().getCommittee();
-        _assignRewardsToCommittee(complianceCommittee, complianceCommitteeAnnualBootstrap);
-
-        lastPayedAt = now;
-    }
-
-    function _assignRewardsToCommittee(address[] memory committee, uint256 annualBootstrapRewards) private {
+    function _assignRewards(address[] memory committee, bool[] memory compliance) private {
         if (committee.length > 0) {
+            uint nCompliance = 0;
+            for (uint i = 0; i < committee.length; i++) {
+                if (compliance[i]) nCompliance++;
+            }
+
             uint256 duration = now.sub(lastPayedAt);
-            uint256 amountPerValidator = Math.min(annualBootstrapRewards.mul(duration).div(365 days), pool.div(committee.length));
-            pool = pool.sub(amountPerValidator * committee.length);
+            uint256 amountPerGeneralValidator = Math.min(generalCommitteeAnnualBootstrap.mul(duration).div(365 days), pool.div(committee.length));
+            uint256 amountPerCompliantValidator = nCompliance == 0 ? 0 :
+                Math.min(complianceCommitteeAnnualBootstrap.mul(duration).div(365 days), pool.div(nCompliance));
 
-            uint256[] memory assignedRewards = new uint256[](committee.length);
+            pool = pool.sub(amountPerGeneralValidator * committee.length).sub(amountPerCompliantValidator * nCompliance);
```

Comment:
Bug - should substract before second Math.min, or sum both elements and then do Math.min